src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java

```diff
@@ -771,23 +776,80 @@ public void onPlayerChangePlotEvent(PlayerChangePlotEvent event) {
 		}		
 	}
 	
-	/*
+	/**
 	 * onOutlawEnterTown
-	 * - Shows message to outlaws entering towns in which they are considered an outlaw.
-	 */
+	 * - Handles outlaws entering a town they are outlawed in.
+	 **/
 	@EventHandler(priority = EventPriority.NORMAL)
 	public void onOutlawEnterTown(PlayerEnterTownEvent event) throws NotRegisteredException {
-		
-		Player player = event.getPlayer();		
-		WorldCoord to = event.getTo();
-		Resident resident = TownyUniverse.getInstance().getDataSource().getResident(player.getName());
 
-		if (to.getTownBlock().getTown().hasOutlaw(resident))
-			TownyMessaging.sendMsg(player, Translation.of("msg_you_are_an_outlaw_in_this_town", to.getTownBlock().getTown()));
+		Resident outlaw = TownyUniverse.getInstance().getDataSource().getResident(event.getPlayer().getName());
+		Town town = event.getEnteredtown();
+		TownBlock location = event.getTo().getTownBlock();
+		
+		String[] split = new String[0];
+		long cooldown = TownySettings.getOutlawKickCooldown();
+		
+		// If this resident is an outlaw in this town
+		if (location.getTown().hasOutlaw(outlaw)) {
+			if (!TownySettings.canOutlawsEnterTowns()) {
+				// Are towns warned when there is an outlaw entering?
+				if (TownySettings.doTownsGetWarnedOnOutlaw()) {
+					TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_outlaw_town_notify").replace("%s", outlaw.getName()));
+				}
+				// Cooldown messages
+				if (cooldown < 1) {
+					TownyMessaging.sendMsg(outlaw, Translation.of("msg_outlaw_kick", town));
+				} else {
```

Comment:
This doesn't need to be there, as the message is shown below when they get teleported which will be right away if the cooldown is 0.

Reply:
Deleted this on the latest commit